shell_zsh.go

```diff
@@ -11,7 +11,11 @@ _direnv_hook() {
 }
 typeset -ag precmd_functions;
 if [[ -z ${precmd_functions[(r)_direnv_hook]} ]]; then
```

Comment:
just thinking about it; if _direnv_hook is already in the array, it should be removed and pushed back. That way it would increase the likelihood that _direnv_hook is in the front of the array.

Reply:
An easy solution would be defining the hook function with `typeset -agU precmd_functions`. This forces only the 1st occurrence of an item is kept in the array.

```zsh
typeset -ag precmd_functions
# re-prepend the item into the array
precmd_functions=( _direnv_hook ${precmd_functions[@]} )
```

But do we really need it? Most users init `direnv` with `eval "$(direnv hook zsh)"`.  I don't think a common user will meddle with `precmd` hook manually. The code here is enough to make sure `_direnv_hook` is hooked on `precmd` 1st.

Reply:
It would be an improvement but not part of the original PR's intent so leaving it for now. Thanks!